### PR TITLE
Create message queue for unresolved messages in WS client

### PR DESCRIPTION
**Pull request by pcriadoperez**
This PR introduce a message queue in the client resolver to automatically return messages when a future is created if there are any unresolved messages
This is to avoid the race condition of having a message be resolved between two watch calls.

Tested by calling watch function in each language ✅ 
Test by running `tsx ts/src/pro/test/base/test.watchWithDelay.ts` ✅ 
Test by running `tsx ts/src/pro/test/base/test.client.resolve.ts ` ✅

### CODE REVIEW

**Comment by carlosmiei** [comment]
image

add and update is not available at netstandard2.0/2.1

File: cs/ccxt/ws/Client.cs [hunk 1]
```diff
@@ -75,7 +75,19 @@ public WebSocketClient(string url, string proxy, handleMessageDelegate handleMes
         public Future future(object messageHash2)
         {
             var messageHash = messageHash2.ToString();
-            return (this.futures as ConcurrentDictionary<string, Future>).GetOrAdd (messageHash, (key) =>  new Future());
+            var future = (this.futures as ConcurrentDictionary<string, Future>).GetOrAdd(messageHash, (key) => new Future());
+            if ((this.rejections as ConcurrentDictionary<string, object>).TryRemove(messageHash, out object rejection))
```
Comment (carlosmiei) [inline cs/ccxt/ws/Client.cs:79]: @pcriadoperez can't we just remove it from the queue like this? ⬆️

Comment (pcriadoperez) [reply cs/ccxt/ws/Client.cs:79]: not sure i understood the comment, you mean when we remove from the messageQueue?

Comment (carlosmiei) [reply cs/ccxt/ws/Client.cs:79]: I replaced the addAndUpdate call with this one because it was not supported, I'm just asking if this is enough

Comment (pcriadoperez) [reply cs/ccxt/ws/Client.cs:79]: I checked the docs and I believe it is supported, the problem was it was missing an argument, that I thought was optional. I updated in the latest commit, and was passing the build locally.

Comment (carlosmiei) [reply cs/ccxt/ws/Client.cs:79]: oh ok let s see how the build goes then

**Comment by carlosmiei** [comment]
@pcriadoperez can we add the option?

**Pull request merged**